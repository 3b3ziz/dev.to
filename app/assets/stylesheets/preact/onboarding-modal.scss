.onboarding-body {
  background-color: white;
  position: fixed;
  margin-top: 50px;
  bottom: 0;
  height: 100vh;
  width: 100%;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  line-height: 200%;
  z-index: 11;
  margin: 0;
  background-image: url('/assets/onboarding-background-white.png');
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  background-position: center; /* Center the image */
  background-repeat: no-repeat; /* Do not repeat the image */

  a {
    color: #2e3b9e;
  }
  .sticker-logo {
    height: 80px;
    transform: rotate(-5deg);
    padding: 10px;
    vertical-align: middle;
    top: -10px;
    padding-top: 5px;
  }
  .onboarding-content {
    display: flex;
    flex-direction: row;
    width: 800px;
    height: 350px;
    background-color: white;
    border: 8px solid #6b55c9;
    padding: 30px;
    border-radius: 3px;
    z-index: 1;
    font-size: 20px;
    box-shadow: 0 0 8px 2px rgba(0, 0, 0, 0.3), 8px 0px 13px rgba(0, 0, 0, 0.21);
    // box-shadow: var(--theme-container-box-shadow, 1px 1px 0px #c2c2c2);
    align-items: center;
  }

  .background-image {
    z-index: -1;
    position: absolute;
    top: 60px;
    left: 0px;
  }

  h1 {
    text-align: center;
  }
  h2 {
    font-size: 25px;
  }

  button {
    width: 130px;
    padding: 6px 0px;
    height: auto;
    margin: 11px 20px;
    text-align: center;
    font-weight: bold;
    border-radius: 3px;
    border: 2px solid #0a0a0a;
    border: 2px solid var(--theme-top-bar-write-color, #0a0a0a);
    color: #0a0a0a;
    color: var(--theme-top-bar-write-color, #0a0a0a);
    font-family: 'HelveticaNeue-CondensedBold', 'HelveticaNeueBoldCondensed',
      'HelveticaNeue-Bold-Condensed', 'Helvetica Neue Bold Condensed',
      'HelveticaNeueBold', 'HelveticaNeue-Bold', 'Helvetica Neue Bold',
      'HelveticaNeue', 'Helvetica Neue', 'TeXGyreHerosCnBold', 'Helvetica',
      'Tahoma', 'Geneva', 'Arial Narrow', 'Arial', sans-serif;
    -webkit-appearance: none;
    font-stretch: condensed;
    font-size: 26px;
  }

  .next-button {
    background: #66e2d5;
    background: var(--theme-top-bar-write-background, #66e2d5);
    float: right;
  }

  .about {
    width: 100%;
    label,
    input,
    textarea {
      display: block;
      width: 95%;
      margin: 0 auto;
    }

    input,
    textarea {
      padding: 12px;
      font-size: 19px;
      text-emphasis: bold;
      border: 1px solid #dee6e9;
      border-radius: 3px;
    }

  }

  .checkbox-slide {
    label {
      display: block;
    }

    input[type='checkbox'] {
      -webkit-appearance: none;
      width: 19px;
      height: 19px;
      background: white;
      border-radius: 5px;
      border: 2px solid #555;
      vertical-align: middle;
      margin: 8px;
    }
    input[type='checkbox']:checked {
      background: var(--theme-top-bar-write-background, #66e2d5);
    }
  }

  .warning-message {
    background-color: red;
    opacity: 0.7;
    padding: 10px;
    width: 100%;
    border-radius: 3px;
    color: white;
  }

  .tag {
    padding: 4px 10px 6px;
    font-size: 16px;
    margin: 14px;
    border-radius: 2px;
    font-weight: bold;
    border: none;
    outline: none;
    overflow: hidden;
    text-overflow: ellipsis;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
      Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
      'Segoe UI Symbol';
  }

  .badge-image {
    position: absolute;
  }
  @media screen and (max-width: 800px) {
    .onboarding-content {
      width: 100%;
      height: 100%;
      background-image: url('/assets/mobile-onboarding-background.png');
      background-size: cover;
      background-position: center; /* Center the image */
      background-repeat: no-repeat; /* Do not repeat the image */
      border: none;
    }

    span {
      display: block;
    }

    .back-button {
      background-color: white;
    }
  }
  @media screen and (max-width: 400px) {
    .onboarding-content {
      font-size: 16px;
    }

    button {
      width: 100px;
      font-size: 18px;
    }
  }

  @media screen and (max-width: 330px) {
    button {
      width: 80px;
      font-size: 14px;
    }

    .sticker-logo {
      height: 40px;
    }

    h1 {
      font-size: 20px;
    }
  }
}
