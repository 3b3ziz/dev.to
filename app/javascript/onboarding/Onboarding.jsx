import 'preact/devtools';
import { h, Component } from 'preact';

import WelcomeSlide from './components/WelcomeSlide';
import PersonalInfoForm from './components/PersonalInfoForm';
import EmailListTermsConditionsForm from './components/EmailListTermsConditionsForm';
import ClosingSlide from './components/ClosingSlide';
import FollowTags from './components/FollowTags';
import FollowUsers from './components/FollowUsers';
import BioForm from './components/BioForm';

export default class Onboarding extends Component {
  constructor(props) {
    super(props);

    this.nextSlide = this.nextSlide.bind(this);
    this.prevSlide = this.prevSlide.bind(this);

    const slides = [
      WelcomeSlide,
      EmailListTermsConditionsForm,
      BioForm,
      PersonalInfoForm,
      FollowTags,
      FollowUsers,
      ClosingSlide,
    ];

    this.slides = slides.map(SlideComponent => (
      <SlideComponent next={this.nextSlide} prev={this.prevSlide} />
    ));

    this.state = {
      currentSlide: 0,
    };
  }

  nextSlide() {
    const { currentSlide } = this.state;
    const nextSlide = currentSlide + 1;
    if (nextSlide < this.slides.length) {
      this.setState({
        currentSlide: nextSlide,
      });
    }
  }

  prevSlide() {
    const { currentSlide } = this.state;
    const prevSlide = currentSlide - 1;
    if (prevSlide >= 0) {
      this.setState({
        currentSlide: prevSlide,
      });
    }
  }

  render() {
    const { currentSlide } = this.state;
    return <div className="onboarding-body">{this.slides[currentSlide]}</div>;
  }
}
