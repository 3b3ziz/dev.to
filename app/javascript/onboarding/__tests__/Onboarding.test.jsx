import { h, render as preactRender } from 'preact';
// import render from 'preact-render-to-json';
import { shallow, deep } from 'preact-render-spy';
// import { JSDOM } from 'jsdom';
import Onboarding from '../Onboarding';

describe('<Onboarding />', () => {
  it('should move to the next slide upon clicking the next button', () => {
    const onboardingSlides = deep(<Onboarding />);
    onboardingSlides.find('.next-button').simulate('click');
    expect(onboardingSlides.state().currentSlide).toBe(1);
  });

  it('should move to the previous slide upon clicking the back button', () => {
    const onboardingSlides = deep(<Onboarding />);
    onboardingSlides.setState({ currentSlide: 1 });
    onboardingSlides.find('.back-button').simulate('click');
    expect(onboardingSlides.state().currentSlide).toBe(0);
  });

  it('previous slide button should not move backwards past zero', () => {
    const onboardingSlides = deep(<Onboarding />);
    onboardingSlides.find('.back-button').simulate('click');
    expect(onboardingSlides.state().currentSlide).toBe(0);
  });

  it('next slide button should not move past the number of slides', () => {
    const onboardingSlides = deep(<Onboarding />);
    const numberOfSlides = onboardingSlides.slides - 1;
    onboardingSlides.setState({ currentSlide: numberOfSlides });
    onboardingSlides.find('.next-button').simulate('click');
    expect(onboardingSlides.state().currentSlide).toBe(numberOfSlides);
  });
});