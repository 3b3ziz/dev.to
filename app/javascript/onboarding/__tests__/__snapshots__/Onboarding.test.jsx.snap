// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Onboarding /> BioForm renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="onboarding-body">
  <div class="onboarding-main">
    <div class="onboarding-content about">
      <h1>About You!</h1>
      <form>
        <label htmlFor="summary">
          Tell the community about yourself! Write a quick bio about what you do, what you're interested in, or anything else!
          <textarea
            name="summary"
            onChange={[Function bound handleChange]}
            maxLength="120"
          >
          </textarea>
        </label>
      </form>
    </div>
    <nav class="onboarding-navigation">
      <button
        onClick={[Function bound prevSlide]}
        class="back-button"
        type="button"
      >
        BACK
      </button>
      <button
        onClick={[Function bound onSubmit]}
        class="next-button"
        type="button"
      >
        Continue
      </button>
    </nav>
  </div>
</div>

`;

exports[`<Onboarding /> CloseSlide renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="onboarding-body">
  <div class="onboarding-main">
    <div class="onboarding-content">
      <h1>
        Congrats 
        <span
          role="img"
          aria-label="tada"
        >
          ðŸŽ‰
        </span>
         youâ€˜re officially part of the DEV community!
      </h1>
      <ul>
        <li>
          Join the 
          <a href="/welcome">Welcome Thread</a>
        </li>
        <li>
          <a href="/new">Write</a>
           your own DEV post
        </li>
        <li>
          <a href="/top/infinity">Read</a>
           some of the most popular all time posts
        </li>
        <li>
          <a href="/settings">Customize</a>
           your profile
        </li>
      </ul>
    </div>
    <nav class="onboarding-navigation">
      <button
        onClick={[Function bound prevSlide]}
        class="back-button"
        type="button"
      >
        BACK
      </button>
    </nav>
  </div>
</div>

`;

exports[`<Onboarding /> EmailTermsConditionsForm renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="onboarding-body">
  <div class="onboarding-main">
    <div class="onboarding-content checkbox-slide">
      <h2>Some things to check off!</h2>
      <form>
        <label htmlFor="checked_code_of_conduct">
          <input
            type="checkbox"
            name="checked_code_of_conduct"
            id="checked_code_of_conduct"
            checked={false}
            onChange={[Function bound handleChange]}
           />
          You agree to uphold our 
          <a href="/code-of-conduct">Code of Conduct</a>
        </label>
        <label htmlFor="checked_terms_and_conditions">
          <input
            type="checkbox"
            id="checked_terms_and_conditions"
            name="checked_terms_and_conditions"
            checked={false}
            onChange={[Function bound handleChange]}
           />
          You agree to our 
          <a href="/terms">Terms and Conditions</a>
        </label>
        <h3>Email Preferences</h3>
        <label htmlFor="email_membership_newsletter">
          <input
            type="checkbox"
            name="email_membership_newsletter"
            checked={true}
            onChange={[Function bound handleChange]}
           />
          Do you want to receive our weekly newsletter emails?
        </label>
        <label htmlFor="email_digest_periodic">
          <input
            type="checkbox"
            name="email_membership_newsletter"
            checked={true}
            onChange={[Function bound handleChange]}
           />
          Do you want to receive a periodic digest with some of the top posts from your tags?
        </label>
      </form>
    </div>
    <nav class="onboarding-navigation">
      <button
        onClick={[Function bound prevSlide]}
        class="back-button"
        type="button"
      >
        BACK
      </button>
      <button
        onClick={[Function bound onSubmit]}
        class="next-button"
        type="button"
      >
        Continue
      </button>
    </nav>
  </div>
</div>

`;

exports[`<Onboarding /> FollowTags renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="onboarding-body">
  <div class="onboarding-main">
    <div class="onboarding-content">
      <h2>Follow some tags!</h2>
      <div class="scroll">
        <button
          type="button"
          onClick={[Function onClick]}
          style="background-color: #000000; color: #ffffff; opacity: 1;"
          class="tag"
        >
          #discuss
        </button>
        <button
          type="button"
          onClick={[Function onClick]}
          style="background-color: #f7df1e; color: #000000; opacity: 1;"
          class="tag"
        >
          #javascript
        </button>
        <button
          type="button"
          onClick={[Function onClick]}
          style="background-color: #2a2566; color: #ffffff; opacity: 1;"
          class="tag"
        >
          #career
        </button>
      </div>
    </div>
    <nav class="onboarding-navigation">
      <button
        onClick={[Function bound prevSlide]}
        class="back-button"
        type="button"
      >
        BACK
      </button>
      <button
        onClick={[Function bound handleComplete]}
        class="next-button"
        type="button"
      >
        Continue
      </button>
    </nav>
  </div>
</div>

`;

exports[`<Onboarding /> FollowUsers renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="onboarding-body">
  <div class="onboarding-main">
    <div class="onboarding-content">
      <h2>Follow some users!</h2>
      <div class="scroll">
        <button
          type="button"
          style="background-color: white;"
          onClick={[Function onClick]}
          class="user"
        >
          <img
            src="ben.jpg"
            alt=""
           />
          Ben Halpern
        </button>
        <button
          type="button"
          style="background-color: white;"
          onClick={[Function onClick]}
          class="user"
        >
          <img
            src="clown.jpg"
            alt=""
           />
          Krusty the Clown
        </button>
        <button
          type="button"
          style="background-color: white;"
          onClick={[Function onClick]}
          class="user"
        >
          <img
            src="dev.jpg"
            alt=""
           />
          dev.to staff
        </button>
      </div>
    </div>
    <nav class="onboarding-navigation">
      <button
        onClick={[Function bound prevSlide]}
        class="back-button"
        type="button"
      >
        BACK
      </button>
      <button
        onClick={[Function bound handleComplete]}
        class="next-button"
        type="button"
      >
        Continue
      </button>
    </nav>
  </div>
</div>

`;

exports[`<Onboarding /> IntroSlide renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="onboarding-body">
  <div class="onboarding-main">
    <div class="onboarding-content">
      <h1>
        <span>Welcome to the </span>
        <img
          src="/assets/purple-dev-logo.png"
          class="sticker-logo"
          alt="DEV"
         />
        <span>community!</span>
      </h1>
      <p>
        DEV is where programmers share ideas and help each other grow. Itâ€™s a global community for contributing and discovering great ideas, having debates, and making friends.
      </p>
      <p>
        A couple quick questions for you before you get started...
      </p>
    </div>
    <nav class="onboarding-navigation">
      <button
        onClick={[Function bound onSubmit]}
        class="next-button"
        type="button"
      >
        Continue
      </button>
    </nav>
  </div>
</div>

`;

exports[`<Onboarding /> PersonalInformationForm renders properly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="onboarding-body">
  <div class="onboarding-main">
    <div class="onboarding-content about">
      <h1>About You!</h1>
      <form>
        <label htmlFor="location">
          Where are you located?
          <input
            type="text"
            name="location"
            id="location"
            onChange={[Function bound handleChange]}
            maxLength="60"
           />
        </label>
        <label htmlFor="employment_title">
          What is your title?
          <input
            type="text"
            name="employment_title"
            id="employment_title"
            onChange={[Function bound handleChange]}
            maxLength="60"
           />
        </label>
        <label htmlFor="employer_name">
          Where do you work?
          <input
            type="text"
            name="employer_name"
            id="employer_name"
            onChange={[Function bound handleChange]}
            maxLength="60"
           />
        </label>
      </form>
    </div>
    <nav class="onboarding-navigation">
      <button
        onClick={[Function bound prevSlide]}
        class="back-button"
        type="button"
      >
        BACK
      </button>
      <button
        onClick={[Function bound onSubmit]}
        class="next-button"
        type="button"
      >
        Continue
      </button>
    </nav>
  </div>
</div>

`;
